programs/dialect/src/lib.rs

```diff
@@ -4,15 +4,15 @@
 //!
 //! The entrypoints and data structures below implement the one-on-one messaging thread use case, as well as associated authentication and management of such threads.
 
-use std::array;
-
 use anchor_lang::prelude::*;
+use solana_program::entrypoint::ProgramResult;
```

Comment:
Anchor 0.23 switched to using a new type alias, `Result<()>` instead of ProgramResult, but for now I'm just re-importing the old way.